src/server/main_service.cc

```diff
@@ -590,6 +600,42 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   GenericFamily::Init(&pp_);
   server_family_.Init(acceptor, std::move(listeners), &cluster_family_);
 
+  if (string flag = GetFlag(FLAGS_replicaof); !flag.empty()) {
+    string ip, port;
+    {
+      auto pos = flag.find_last_of(':');
+      CHECK(pos != string::npos) << "Invalid format for --replicaof: missing ':'.";
+
+      string ip1 = flag.substr(0, pos);
+      port = flag.substr(pos + 1);
+
+      CHECK(!ip1.empty() && !port.empty())
+          << "Invalid format for --replicaof: IP or port are empty.";
+
+      // For IPv6
+      CHECK(!((ip1.front() == '[') ^ (ip1.back() == ']')))
+          << "Invalid format for --replicaof: unclosed brackets.";
```

Comment:
Wow... that's a complex expression, especially with that bitwise XOR :)
How about the below?
```cpp
CHECK(ip1.front() != '[' || ip1.back() == ']') << ...;
```

Reply:
Oops! Your check is not exactly the same. 
The condition I was aiming for is `ip1.front() == '['` *iff* `ip1.back() == ']'`. Maybe its complex but this is the most efficient way of computing iff (XNOR).

Reply:
I realize it's not exactly the same, but you don't hermetically check the contents of `ip1` anyway (like if it has `[]` inside it, or other invalid cases).
I thought that this expression can be simplified without losing functionality (as the connection will break anyway), but your call if you think it's important :)